### PR TITLE
Adding telemetry to storage dataplane

### PR DESCRIPTION
**Pull request by mcardosos**
Taking some cool ideas from the [Ruby SDK telemetry](https://github.com/Azure/azure-sdk-for-ruby/pull/543), users can append more info to the User Agent using the `AddToUserAgent` func.
The User-Agent header follows this format...
```
Go/[version] ([arch]-[OS]) Azure-SDK-for-Go/[version] storage-dataplane/[APIversion] [storageService]
```
... and would look like this...
```
Go/go1.7 (amd64-windows) Azure-SDK-for-Go/7.0.1-beta storage-dataplane/2015-02-21 queue
```

PTAL
@salameer @kirthik @vishrutshah @marstr @jhendrixMSFT

### CODE REVIEW

File: storage/version.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package storage
+
+var (
+	sdkVersion = "7.0.1-beta"
```
Comment (jhendrixMSFT) [inline storage/version.go]: I believe we need to update gen.go to automatically update this version yes?

**Comment by jhendrixMSFT** [comment]
The changes in file.go will likely be unnecessary when the refactoring changes in the dev branch get merged.

File: storage/version.go (hunk 1 above)
Comment (mcardosos) [inline storage/version.go]: Fixed

File: storage/client.go [hunk 2]
```diff
@@ -155,13 +159,41 @@ func NewClient(accountName, accountKey, blobServiceBaseURL, apiVersion string, u
...
 		baseURL:     blobServiceBaseURL,
 		apiVersion:  apiVersion,
-	}, nil
+	}
+	c.userAgent = c.getDefaultUserAgent()
+	return c, nil
+}
+
+func (c Client) getDefaultUserAgent() string {
+	return fmt.Sprintf("Go/%s (%s-%s) Azure-SDK-for-Go/%s storage-dataplane/%s",
```
Comment (mcardosos) [inline storage/client.go]: Fixed (we need to change AutoRest to gen the UserAgent for ARM with F too)

**Comment by mcardosos** [comment]
@jhendrixMSFT I merged the changes from dev branch, and changed the branch to merge to dev too

**Pull request merged**